src/Bicep.Core/Analyzers/Linter/ApiVersions/ApiVersionProvider.cs

```diff
@@ -69,17 +67,11 @@ private ApiVersionCache EnsureCached(ResourceScope scope)
             }
             cache.typesCached = true;
 
-            IEnumerable<ResourceTypeReference> resourceTypeReferences;
-            if (cache.injectedTypes is null)
-            {
-                resourceTypeReferences = resourceTypeLoader.GetAvailableTypes();
-            }
-            else
-            {
-                resourceTypeReferences = cache.injectedTypes;
-            }
+             var resourceTypesToCache = cache.injectedTypes is null
```

Comment:
cache.injectedTypes is no longer needed, right?

Reply:
honestly, I don't think I understand well how its used and wether its prescence is needed. I am totally up for refactoring it out but I think that should be done as a separate PR so that we don't both add behavior and refactor unless strictly necessary. What do you think?